tests/sentry/issues/endpoints/test_organization_group_index.py

```diff
@@ -2647,6 +2656,87 @@ def test_snuba_assignee_filter(self):
                 assert int(response.data[2]["id"]) == event7.group.id
                 assert int(response.data[3]["id"]) == event4.group.id
                 assert int(response.data[4]["id"]) == event3.group.id
+            # Negated queries
+            elif query == "!assigned_or_suggested:[me]":
+                assert len(response.data) == 7
+                assert int(response.data[0]["id"]) == event9.group.id
+                assert int(response.data[1]["id"]) == event8.group.id
+                assert int(response.data[2]["id"]) == event7.group.id
+                assert int(response.data[3]["id"]) == event6.group.id
+                assert int(response.data[4]["id"]) == event5.group.id
+                assert int(response.data[5]["id"]) == event4.group.id
+                assert int(response.data[6]["id"]) == event2.group.id
```

Comment:
Maybe we should store the expected results alongside the query, so you can do
```
for query, expected_group_ids in [("query", [event9.group.id, ...]), ...]:
            response = self.get_success_response(
                sort="new",
                useGroupSnubaDataset=1,
                query=query,
            )
            assert [row["id"] for row in response.data] == expected_group_ids
    ```

Reply:
@wedamija Good catch! And cursor was able to take your suggestion and refactor all the code in one go, amazing!